sklearn/utils/estimator_checks.py

```diff
@@ -319,15 +402,22 @@ def check_dtype_object(name, Estimator):
     assert_raises_regex(TypeError, msg, estimator.fit, X, y)
 
 
-def check_transformer_general(name, Transformer):
-    X, y = make_blobs(n_samples=30, centers=[[0, 0, 0], [1, 1, 1]],
-                      random_state=0, n_features=2, cluster_std=0.1)
-    X = StandardScaler().fit_transform(X)
-    X -= X.min()
+def check_transformer_general(name, Transformer, readonly=False):
+    X, y = _make_blobs_with_mode(n_samples=30, centers=[[0, 0, 0], [1, 1, 1]],
+                                 random_state=0, n_features=2, cluster_std=0.1,
+                                 readonly=readonly, positive=True, scale=True)
+    # Put into _make_blobs :
+    # X -= X.min()
+    # X = StandardScaler().fit_transform(X)
```

Comment:
Please remove those commented out line now that they have been reimplemented by the mean of the newly introduced `positive=True`  option.